core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -234,6 +239,8 @@ private abstract static class AbstractRequestMetrics implements RequestMetrics {
 
             requestDuration = newTimer(parent, idPrefix.name("request.duration"), idPrefix.tags());
             requestLength = newDistributionSummary(parent, idPrefix.name("request.length"), idPrefix.tags());
+            requestFullyReceivedDuration = newTimer(parent, idPrefix.name("request.received.duration"),
```

Comment:
I think this property should be in `DefaultServiceRequestMetrics` because it is a server-side property. Could move it?

Reply:
I tried but if I moved it to `DefaultServiceRequestMetrics`,  `DefaultClientRequestMetrics` also needs to implement it.
Or if I totally remove it from `RequestMetric` interface, [a metric recording logic](https://github.com/line/armeria/pull/5021/files#diff-bc19bd339166e0240aad34cc1d72a4cffa1b55140064f9da31381db924cb19f3R152) doesn't work. 

That's why I added it here. Any idea?
@minwoox

Reply:
Can't we just add it to `ServiceRequestMetrics` and record it?
```java
private interface ServiceRequestMetrics extends RequestMetrics {
    Counter requestTimeouts();
    Timer requestReceivedDuration();
}
...
if (server) {
    final ServiceRequestMetrics metrics = MicrometerUtil.register(registry, idPrefix,
                                                                  ServiceRequestMetrics.class,
                                                                  DefaultServiceRequestMetrics::new);
    updateMetrics(log, metrics, isSuccess);
    metrics.requestReceivedDuration().record(...);
    if (log.responseCause() instanceof RequestTimeoutException) {
        metrics.requestTimeouts().increment();
    }
    return;
}
```

Reply:
Great idea 👍